### PR TITLE
gnum4: build html and plain text documentation

### PR DESCRIPTION
**Pull request by KAction**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by KAction** [comment]
I built html and plaintext because I want these formats myself. Probably we need more principled solution that would include all formats supported by texinfo.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 136042` at 065a0bff run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>m4-doc (gnum4-doc)</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/tools/misc/gnum4/doc.nix:10:3:
```
   |
10 |   buildPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/development/tools/misc/gnum4/doc.nix:10:3:
```
   |
10 |   buildPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/misc/gnum4/doc.nix:14:3:
```
   |
14 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md">license-missing</a>

Package is missing a license.

Near pkgs/development/tools/misc/gnum4/doc.nix:10:3:
```
   |
10 |   buildPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/tools/misc/gnum4/doc.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{ stdenv, gnum4, texinfo4 }:
+stdenv.mkDerivation {
+  inherit (gnum4) src version;
+  pname = "gnum4-doc";
+  outputs = [ "out" "html" "txt" ];
+  phases = [ "unpackPhase" "buildPhase" "installPhase" ];
```
Comment (veprbl) [inline pkgs/development/tools/misc/gnum4/doc.nix:6]: Override `phases` is discouraged as it implicitly disables some less-known but useful phases. Attributes like `dontConfigure` are to be used instead.

**Review by veprbl — COMMENTED** [review]
When implemented as a separate `m4-doc` package, this is not easily discoverable. It should be possible to add an optional argument (`isBootstrap ? false` or `withManual ? true`) to `gnum4` to build without documentation in bootstrap and with documentation in the final package set.

File: pkgs/development/tools/misc/gnum4/doc.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  buildPhase = ''
+    makeinfo --html --no-split doc/m4.texi
+    makeinfo --plaintext --no-split doc/m4.texi > m4.txt
+  '';
+  installPhase = ''
+    mkdir $out
+    mkdir -p {$txt,$html}/share/doc/m4
+    cp ./m4.html $html/share/doc/m4
+    cp ./m4.txt $txt/share/doc/m4
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/gnum4/doc.nix:19]: Missing meta

**Pull request closed without merging**